### ISSUE TITLE
`--confirm` not respected when deploying zarf package

### ISSUE THREAD

**Issue by RothAndrew** [defenseunicorns/zarf#200]
**What I saw:**

If I deploy the Big Bang example, which has an optional component called `kubescape`, it prompts me to choose whether I want Kubescape or not

**What I expected:**

If I run `zarf package deploy foobar.tar.gz --confirm` I expect that it would never stop and prompt me to interact, because I used the `--confirm` flag. It would either make a choice itself or fatal out if it cant. If the `--components` flag was not used I'm thinking it should use the defaults that are set for each component.

**Comment by RothAndrew** [comment]
Note: This is an existing finding from before the big PR from Friday. It's not an urgent priority before we cut the next release.

**Comment by RothAndrew** [comment]
I'm also curious how we would non-interactively deploy the big bang example without kubescape. Since it is the only optional component omitting it would be the same as asking for the defaults, which in this case is "yes". I wonder if we had Zarf look for an empty string or something to decide that the user is saying "I don't want any of the components that are optional.

Possible example:
```
# option 1
zarf package deploy foobar.tar.zst --confirm --components ""
# option 2
zarf package deploy foobar.tar.zst --confirm --no-optional-components
# option 3
???
```

**Comment by jeff-mccoy** [comment]
> Note: This is an existing finding from before the big PR from Friday. It's not an urgent priority before we cut the next release.

Thanks for clarifying I was about to 🤦‍♂️

**Comment by RothAndrew** [comment]
#271 should fix this. It's getting merged into #237, so when #237 merges this should get resolved.
